valhalla/proposals/accounting.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import requests
+from django.conf import settings
+
+from valhalla.proposals.models import Semester
+
+logger = logging.getLogger(__name__)
+
+
+def split_time(start, end, chunks=4):
+    spans = []
+    chunk = (end - start) / chunks
+    spans.append((start, start + chunk))
+    for i in range(0, chunks - 1):
+        spans.append((spans[i][1], spans[i][1] + chunk))
+    return spans
+
+
+def get_time_totals_from_pond(timeallocation, start, end, too, recur=0):
+    """
+    Pond queries are too slow with large proposals and time out, this hack splits the query
+    when a timeout occurs
+    """
+    if recur > 3:
+        raise RecursionError('Pond is timing out, too much recursion.')
+
+    total = 0
+    try:
+        total += query_pond(timeallocation.proposal.id, start, end, timeallocation.telescope_class, too)
+    except requests.HTTPError:
+        logger.warning('We got a pond inception. Splitting further.')
```

Comment:
It will stop execution at any time if the recursion depth is greater than 4. See: 

https://github.com/LCOGT/valhalla/blob/8a724ffb16b95fcd8d819e529af8e1888fcac613/valhalla/proposals/tests.py#L106

I don't think it will give wrong answers if the pond comes back during the recursive call. The the scenario below if the pond is down, we split, it's still down, but then comes back.

Try Query ->
    Exception, Split -> 
         Try Query 1->
            Exception, Split ->
                Try Query 1.1
                    success, add 1.1
                 Try Query 1.2
                     success, add 1.2
                 Try Query 1.3
                      success, add 1.3 
                 Try Query 1.4
                      success, add 1.4   
         Try Query 2 ->
              success, add 2
         Try Query 3
              success, add 3
         Try Query 4
              success, add 4